pdf_rendering.py

```diff
@@ -1,239 +1,378 @@
+            pdf.set_x(pdf.get_x() + 5)
+            pdf.cell(w=table_column_client_code_width, txt=f'{task.project_id}', align = 'C')
 
-        pdf.set_x(pdf.get_x()+5)
+            pdf.set_x(pdf.get_x() + 5)
 
-        processedDescription = process_description(pdf,desc)
-        current_y = pdf.get_y()
-        for line in processedDescription:
-            current_x = pdf.get_x()
-            pdf.cell(w=tableColumnDescriptionWidth,txt=f'{line}', align = 'L', ln=1)
-            pdf.set_y(pdf.get_y()+5)
-            pdf.set_x(current_x)
+            processed_description = process_text_to_fit_width(pdf, task.name, text_width)
+            current_y = pdf.get_y()
+            for line in processed_description:
+                current_x = pdf.get_x()
+                pdf.cell(w=table_column_description_width, txt=f'{line}', align = 'L', ln=1)
+                pdf.set_y(pdf.get_y() + 5)
+                pdf.set_x(current_x)
                 
-            pdf.set_y(current_y)
-            pdf.set_x(pdf.get_x() + tableColumnDescriptionWidth + 85)
-            pdf.cell(w=tableColumnHoursWidth,txt=f'{duration}',align = 'C')
-            pdf.set_y(pdf.get_y() + len(processedDescription)*5)
+                pdf.set_y(current_y)
+                pdf.set_x(pdf.get_x() + table_column_description_width + 85)
+                pdf.cell(w=table_column_hours_width, txt=f'{task.duration}', align = 'C')
+                pdf.set_y(pdf.get_y() + len(processed_description) * 5)
+        
+        pdf.line(initialX + 10, pdf.get_y() - 3, length, pdf.get_y() - 3)
 
-    pdf.line(initialX+10,pdf.get_y()-3,length,pdf.get_y()-3)
-    
-def process_description(pdf,text):
-    words = str(text).split()
-    rows = []
-    row = ''
-    for word in words:
-        next_row = ' '.join([row, word])
-        isLast = False
-        isFitting = False
-
-        if  pdf.get_string_width(next_row) <= 83.50:
-            isFitting = True
-        if word == words[-1]:
-            isLast = True
-
-        if isFitting:
-            row = next_row
-        if isLast or not isFitting:
-            rows.append(row)
-            row = ''
-    return rows
-
-def create_total(pdf,
-                invoice,
-                tableColumnDateWidth, 
-                tableColumnClientCodeWidth,
-                tableColumnDescriptionWidth,
-                tableColumnHoursWidth,
-                ):
-    pdf.set_y(pdf.get_y()+5)
-    pdf.set_x(tableColumnDateWidth + tableColumnClientCodeWidth + tableColumnDescriptionWidth+10)
-    pdf.set_font(family='Arial',style='B')
-    pdf.cell(w=20, txt=f'Total',align='C')
-
-    totalhours = 0
-    for task in invoice.activity.tasks:
-        totalhours += int(task.duration)
+    def render_total():
+        pdf.set_y(pdf.get_y() + 5)
+        pdf.set_x(table_column_date_width + table_column_client_code_width + table_column_description_width+10)
+        pdf.set_font(family='Arial', style='B')
```

Comment:
We should avoid changing the font family everywhere, let's find a way to do this in a single time at the beginning of this rendering.

Reply:
The library used for pdf rendering requires explicit family specification at each font style / size change.